quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -307,21 +313,48 @@ public static void validateNonCliConfig(List<String> cliArgs, AbstractCommand ab
                     throw new PropertyException(PropertyMapperParameterConsumer.getErrorMessage(mapper.getFrom(),
                             value, mapper.getExpectedValues(), mapper.getExpectedValues()) + ". From ConfigSource " + configSource.getName());
                 }
+
+                mapper.getDeprecatedMetadata().ifPresent(d -> {
+                    DeprecatedMetadata metadata = (DeprecatedMetadata) d;
+                    StringBuilder sb = new StringBuilder("\t- ");
+                    sb.append(mapper.getFrom());
+                    if (metadata.getNote() != null || !metadata.getNewOptionsKeys().isEmpty()) {
+                        sb.append(":");
+                    }
+                    if (metadata.getNote() != null) {
+                        sb.append(" ");
+                        sb.append(metadata.getNote());
+                        if (!metadata.getNote().endsWith(".")) {
+                            sb.append(".");
+                        }
+                    }
+                    if (!metadata.getNewOptionsKeys().isEmpty()) {
+                        sb.append(" Consider using ");
+                        sb.append(String.join(", ", metadata.getNewOptionsKeys()));
+                        sb.append(".");
+                    }
+                    deprecatedInUse.add(sb.toString());
+                });
             }
         }
 
+        Logger logger = Logger.getLogger(Picocli.class); // logger can't be instantiated in a class field
+
         if (!ignoredBuildTime.isEmpty()) {
-            outputIgnoredProperties(ignoredBuildTime, true, out);
+            outputIgnoredProperties(ignoredBuildTime, true, logger);
         } else if (!ignoredRunTime.isEmpty()) {
-            outputIgnoredProperties(ignoredRunTime, false, out);
+            outputIgnoredProperties(ignoredRunTime, false, logger);
+        }
+
+        if (!deprecatedInUse.isEmpty()) {
+            logger.warn("The following used options are DEPRECATED and will be removed in a future release:\n" + String.join("\n", deprecatedInUse));
```

Comment:
Works as expected (provided `--cache` and `--https-trust-store-password`).

Small improvement would be to get rid of the KC namespace as we don't deprecate any other props. For property config file source the current state is relevant, but for CLI not so much. But, it's only a nitpicking.   
```
2023-12-11 12:10:03,314 WARN  [org.keycloak.quarkus.runtime.cli.Picocli] (main) The following used options are DEPRECATED and will be removed in a future release:
        - kc.https-trust-store-password: Use the System Truststore instead, see the docs for details.
        - kc.proxy: Consider using proxy-headers.
```